active-rfcs/0000-async-component-api.md

```diff
@@ -104,9 +109,23 @@ const AsyncFooWithOptions = defineAsyncComponent({
 
 ## Retry Control
 
-The new `retryWhen` option expects a function that returns a boolean indicating whether the async component should retry when the loader promise rejects. The function receives the rejection error as the argument so it can conditionally retry only on certain types of errors.
+The new `onError` option provides a hook to perform customized retry behavior in case of a loader error:
+
+``` js
+const Foo = defineAsyncComponent({
+  // ...
+  onError(error, retry, fail, attempts) {
+    if (error.message.match(/fetch/) && attempts <= 3) {
+      // retry on fetch errors, 3 max attempts
+      retry()
+    } else {
+      fail()
+    }
+  }
+})
+```
```

Comment:
> @CyberAP that doesn't seem to be much different from `retryWhen`, and it doesn't have the flexibility @Morgul wanted.
> 
> I agree with @backbond87 that a separate `onError` option might be the most straightforward.

I am ok with that idea as well. Could `onError` arguments be contained in a single destructurable object? So there's no need to remember the exact order in which they're passed to a callback.

```suggestion
``` js
const Foo = defineAsyncComponent({
  // ...
  onError({ error, retry, fail, attempts }) {
    if (error.message.match(/fetch/) && attempts <= 3) {
      // retry on fetch errors, 3 max attempts
      retry()
    } else {
      fail()
    }
  }
})
```
```

Reply:
I don't know, the order feels pretty natural to me. With destructure instead of the order, you'll have to remember the exact names. It's not much different imo.

Reply:
Its less about remembering imho, but more about needing only a few of the args most of the time, depending on the use case. See my examples